java/image.bzl

```diff
@@ -114,18 +114,31 @@ def java_files(f):
     Returns:
         Depset of java source files.
     """
-    files = depset()
+    files = []
+
     if java_common.provider in f:
         java_provider = f[java_common.provider]
-        files = depset(transitive = [files, java_provider.transitive_runtime_jars])
-    if hasattr(f, "files"):  # a jar file
-        files = depset(transitive = [files, f.files])
-    return files
+        files.append(java_provider.transitive_runtime_jars)
+
+    f_files = f[DefaultInfo].files
+    if f_files != None:
+        files.append(f_files)
+
+    return depset(transitive = files)
 
 def java_files_with_data(f):
+    """Filter out the list of java source and data files from the given list of runfiles.
+
+    Args:
+       f: Runfiles for a java_image rule.
+
+    Returns:
+       Depset of java source and data files.
+    """
     files = java_files(f)
-    if hasattr(f, "data_runfiles"):
-        files = depset(transitive = [files, f.data_runfiles.files])
+    runfiles = f[DefaultInfo].data_runfiles.files
```

Comment:
I think you wanna check `f[DefaultInfo].data_runfiles` is not None, otherwise f[DefaultInfo].data_runfiles.files will complain None type does not have `files` attribute.

Reply:
oh, yeah. Checked the wrong thing.